newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js

```diff
@@ -71,24 +72,32 @@ export default class DirectionTools extends Component<Props, State> {
 
   saveTimeBetweenFrames = () => {
     const { direction } = this.props;
+    const currentTimeBetweenFrames = direction.getTimeBetweenFrames();
+    const newTimeBetweenFrames = Math.max(
+      parseFloat(this.state.timeBetweenFrames),
+      0.00001
+    );
+    if (currentTimeBetweenFrames === newTimeBetweenFrames) return;
 
-    const newTime = Math.max(parseFloat(this.state.timeBetweenFrames), 0.00001);
-    const newTimeIsValid = !isNaN(newTime);
+    const newTimeIsValid = !isNaN(newTimeBetweenFrames);
 
-    if (newTimeIsValid) direction.setTimeBetweenFrames(newTime);
-    this.setState({
-      timeBetweenFrames: formatTime(
-        this.props.direction.getTimeBetweenFrames()
-      ),
-      timeError: newTimeIsValid,
-    });
+    if (newTimeIsValid) {
+      direction.setTimeBetweenFrames(newTimeBetweenFrames);
+      this.setState({
+        timeBetweenFrames: formatTime(newTimeBetweenFrames),
+        timeError: newTimeIsValid,
```

Comment:
I feel like this timeError is not used: if the time is valid, time error is set to true 😅

Reply:
uh indeed